crates/router/src/connector/adyen.rs

```diff
@@ -224,6 +224,101 @@ impl ConnectorValidation for Adyen {
+                api_models::payments::PayLaterData::KlarnaRedirect { .. } => Ok(()),
+                api_models::payments::PayLaterData::AlmaRedirect {}
+                | api_models::payments::PayLaterData::AtomeRedirect {}
+                | api_models::payments::PayLaterData::KlarnaSdk { .. }
+                | api_models::payments::PayLaterData::AffirmRedirect {}
+                | api_models::payments::PayLaterData::PayBrightRedirect {}
+                | api_models::payments::PayLaterData::WalleyRedirect {}
+                | api_models::payments::PayLaterData::AfterpayClearpayRedirect { .. } => {
+                    mandate_not_supported_error!(pm_type, self.id())
+                }
+            },
+
+            api_models::payments::PaymentMethodData::BankRedirect(bank_redirect) => {
+                match bank_redirect {
+                    api_models::payments::BankRedirectData::Ideal { .. }
+                    | api_models::payments::BankRedirectData::Sofort { .. }
+                    | api_models::payments::BankRedirectData::OpenBankingUk { .. }
+                    | api_models::payments::BankRedirectData::Giropay { .. }
+                    | api_models::payments::BankRedirectData::Trustly { .. }
+                    | api_models::payments::BankRedirectData::BancontactCard { .. } => Ok(()),
+                    api_models::payments::BankRedirectData::OnlineBankingFinland { .. }
+                    | api_models::payments::BankRedirectData::OnlineBankingPoland { .. }
+                    | api_models::payments::BankRedirectData::OnlineBankingSlovakia { .. }
+                    | api_models::payments::BankRedirectData::OnlineBankingFpx { .. }
+                    | api_models::payments::BankRedirectData::OnlineBankingThailand { .. }
+                    | api_models::payments::BankRedirectData::Bizum {}
+                    | api_models::payments::BankRedirectData::Blik { .. }
+                    | api_models::payments::BankRedirectData::Eps { .. }
+                    | api_models::payments::BankRedirectData::Interac { .. }
+                    | api_models::payments::BankRedirectData::Przelewy24 { .. }
+                    | api_models::payments::BankRedirectData::OnlineBankingCzechRepublic {
+                        ..
+                    } => mandate_not_supported_error!(pm_type, self.id()),
+                }
+            }
+
+            api_models::payments::PaymentMethodData::BankDebit(bank_debit) => match bank_debit {
+                api_models::payments::BankDebitData::AchBankDebit { .. }
+                | api_models::payments::BankDebitData::SepaBankDebit { .. }
+                | api_models::payments::BankDebitData::BecsBankDebit { .. } => Ok(()),
+                api_models::payments::BankDebitData::BacsBankDebit { .. } => {
+                    mandate_not_supported_error!(pm_type, self.id())
+                }
+            },
+            api_models::payments::PaymentMethodData::MandatePayment => Ok(()),
```

Comment:
> Do we still need 'MandatePayment'

Short answer - Yes

Currently in authorize flow, the `payment_method_data` is mandatory. But in case of recurring mandates, if the payment method data is not saved in locker ( only connector mandate ids are available ), then the payment method data would have to be empty in `PaymentsAuthorizeData` request, but it cannot be since type system does not allow that. So in order to get around this issue, the `MandatePayment` still should exist.